### PR TITLE
fix auto login

### LINKED ISSUE #205: headless logging in failing
**Describe the bug**

**To Reproduce**
1. run `spotify-dl` with the `--p`, `--u`, and `--l` options
2. after about 30 seconds, this error appears:
```asm
i Generating new access token
i Performing Spotify Auth Please Wait...
× waiting for selector `#auth-accept` failed: timeout 30000ms exceeded
Error: Could not generate token Please find a screenshot of why the auto login failed at  ./failure.png
    at requestAuthorizedTokens (file:///C:/Users/jj/AppData/Roaming/npm/node_modules/spotify-dl/lib/api.js:147:13)
    at async checkCredentials (file:///C:/Users/jj/AppData/Roaming/npm/node_modules/spotify-dl/lib/api.js:86:7)
    at async verifyCredentials (file:///C:/Users/jj/AppData/Roaming/npm/node_modules/spotify-dl/lib/api.js:61:5)
    at async callSpotifyApi (file:///C:/Users/jj/AppData/Roaming/npm/node_modules/spotify-dl/lib/api.js:189:5)
    at async extractSavedTracks (file:///C:/Users/jj/AppData/Roaming/npm/node_modules/spotify-dl/lib/api.js:475:19)
    at async getSavedTracks (file:///C:/Users/jj/AppData/Roaming/npm/node_modules/spotify-dl/util/get-songdata.js:84:10)    at async run (file:///C:/Users/jj/AppData/Roaming/npm/node_modules/spotify-dl/util/runner.js:229:22)
```

**Expected behavior**
it logs in

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by TheBotlyNoob**
# Title
fix auto login

## Details
adds a much more bulletproof selector for the "do you agree" popup

## Testing
it runs correctly using the steps in #205

## Checklist
[] Ran eslint/prettier formatting
[] Tests are passing (no tests yet)
[x] Feature is considered complete

## Issues this resolves
fixed #205

### CODE REVIEW

**Comment by phyzical** [comment]
@TheBotlyNoob thanks for the pr, one thing though

 would this text change if a user was not English? 

(if so an id selector may still be more appropriate)

**Comment by SwapnilSoni1999** [comment]
any updates on this?

**Comment by orowith2os** [comment]
The question @phyzical made still needs addressing.

File: lib/api.js [hunk 1]
```diff
@@ -135,8 +121,9 @@ const requestAuthorizedTokens = async () => {
       await page.type('#login-username', username);
       await page.type('#login-password', password);
       await page.click('#login-button');
-      await page.waitForSelector('#auth-accept');
-      await page.click('#auth-accept');
+      await page
+        .waitForSelector('#auth-accept, *[data-testid="auth-accept"]')
+        .then(e => e.click());
```
Comment (phyzical) [inline lib/api.js:126]: wouldn't this be the same as what were already doing as dom land there is only 1 element with an id of `auth-accept` ?

**Comment by TheBotlyNoob** [comment]
it would not. the reason the autologin broke is that Spotify changed the button to be `id="auth-accept"` to `data-testid="auth-accept"`. This makes it so that the id selector doesn't match the element, breaking it.

**Comment by phyzical** [comment]
Oh Sorry, i did not read that selector properly.

You have just adjusted it to be an OR.

But thanks for the explanation LGTM

**Pull request merged**